src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapXMLUtils.java

```diff
@@ -329,4 +329,18 @@ public static int parseGroupMemberRoleId(final String localRolesForGroupXML) thr
                 u.getChild("Lastname").getText().trim(),
                 u.getChild("Email").getText().trim())).collect(Collectors.toList());
     }
+
+    /**
+     * Generates an XML string needed to create a new group via  the SOAP interface.
+     * @param groupName The name of the group
+     * @return The XML string
+     */
+    public static String generateAddGroupXMLFromName(String groupName) {
+        return "<?xml version=\"1.0\" encoding=\"utf-8\"?>" +
```

Comment:
public boolean addGroup(IliasNode course, String groupName)

Reply:
I have used Stringbuilder initially, but then Intellij suggests to convert it to a String. The situation is also described here: 
https://stackoverflow.com/questions/42855964/how-to-prevent-notice-stringbuilder-sb-can-be-replaced-with-string-in-int 
and here 
https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000093250-Convert-StringBuilder-to-String-concatenation-suggestion.
Thus I followed the suggestion. However, if you still want to, I can of course use a StringBuilder.

Reply:
Just for readability, but let's leave it with concatenation. It's fine